src/modules/extra/m_ssl_openssl.cpp

```diff
@@ -204,6 +204,47 @@ namespace OpenSSL
 			return SSL_CTX_load_verify_locations(ctx, filename.c_str(), 0);
 		}
 
+		void SetCRL(const std::string& crlfile, const std::string& crlpath, const std::string& crlmode)
+		{
+			unsigned long crlflags;
+			X509_STORE* store;
+			if (crlmode == "chain")
+			{
+				crlflags = X509_V_FLAG_CRL_CHECK | X509_V_FLAG_CRL_CHECK_ALL;
+			}
+			else if (crlmode == "leaf")
+			{
+				crlflags = X509_V_FLAG_CRL_CHECK;
```

Comment:
This can be the default value and you can `|= X509_V_FLAG_CRL_CHECK_ALL` in the "chain" block and then you can do != "leaf" instead of using an else block.